packages/neuron-wallet/src/keys/hd.ts

```diff
@@ -10,127 +9,32 @@ enum BIP44Params {
   Account = "0'",
 }
 
-enum AddressType {
-  Receiving = 0,
-  Change = 1,
-}
-
-const MaxAddressNumber = 30
-
 class HD {
-  // Generate both receiving and change addresses
-  public static generateAddresses = (
-    keysData: KeysData,
-    receivingAddressNumber: number,
-    changeAddressNumber: number,
-  ) => {
-    if (receivingAddressNumber < 1 || changeAddressNumber < 1) {
-      throw new Error('Address number error.')
-    } else if (receivingAddressNumber > MaxAddressNumber || changeAddressNumber > MaxAddressNumber) {
-      throw new Error('Address number error.')
-    }
-    const receivingAddresses = []
-    const changeAddresses = []
-    const root: bip32.BIP32Interface = bip32.fromPrivateKey(
-      Buffer.from(keysData.privateKey, 'hex'),
-      Buffer.from(keysData.chainCode, 'hex'),
-    )
-    for (let index = 0; index < receivingAddressNumber; index++) {
-      receivingAddresses.push(HD.addressFromHDIndex(root, index, AddressType.Receiving))
-    }
-    for (let index = 0; index < changeAddressNumber; index++) {
-      changeAddresses.push(HD.addressFromHDIndex(root, index, AddressType.Change))
-    }
-    return {
-      receiving: receivingAddresses,
-      change: changeAddresses,
-    }
+  public static pathFromIndex = (type: AddressType, index: number) => {
+    return `m/${BIP44Params.Purpose}/${BIP44Params.CoinTypeTestnet}/${BIP44Params.Account}/${type}/${index}`
```

Comment:
Is `CoinTypeTestnet` hard coded here?

Reply:
We'll have a fixed number for `coin type` later, but most likely we'll use the same one for both testnet and mainnet.

Reply:
In face we can just hard code this like `m/44/${BIP44Params.CoinTypeTestnet}/0/${type}/${index}`. `44` (purpose) and `0` (account) are well defined by BIP and won't change.

Reply:
`44` (purpose) and `0` (account) are constant. And the reason is that  just a number can't tell others what mean. Maybe `enum` is not a better way to be as a constant. I don't know.

Reply:
And we should avoid magic number according to default address number, so I think we should give every constant a more readable name.

Reply:
So `m/44/.../0/` is not a magic number, it's a de fact standard path.

If we need to define `44` etc. as constant and contact them, then why not also make `m` a constant? Naming and variable decision sometimes need certain degree of balance, not stick rules.

Extend this topic, it's fairly common for blockchain developers to search for `m/44` to look at the HD path inside a project's source code - that won't work for our variable names.

Reply:
OK. I see.